app/src/main/java/com/example/fundamentalskotlin/FragmentMoviesList.kt

```diff
@@ -0,0 +1,56 @@
+package com.example.fundamentalskotlin
+
+import android.content.Context
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import androidx.fragment.app.Fragment
+
+class FragmentMoviesList: Fragment()  {
+
+    private var ivMovies: ImageView? = null
+    var listener: ClickListener? = null
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? = inflater.inflate(R.layout.fragment_movies_list, container, false)
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        ivMovies = view.findViewById<ImageView>(R.id.iv_mask).apply {
+            setOnClickListener { listener?.addFragmentMoviesDetails() }
+        }
+
+    }
+
+    override fun onAttach(context: Context) {
+        super.onAttach(context)
+        if (context is ClickListener){
+            listener = context
+        }
```

Comment:
Можно написать лакончинее
```suggestion
        listener = context as? ClickListener
```